### PR TITLE
fix: text formatting icons in inline menu & slash commands

### PR DESCRIPTION
**Pull request by daibhin**
## Problem

We currently use a big of text & icons for 

## Changes

- Update bold and italic icons to be centered inside the svg
- Remove non existent `text-md` class
- Add ability to toggle headers from inline bubble menu
- Make a custom heading icon for notebooks while we get some from design

## How did you test this code?

Visually

https://github.com/PostHog/posthog/assets/6685876/6cb2c2ca-a606-40f7-bce1-1dc77a32ea66

### CODE REVIEW

File: frontend/src/scenes/notebooks/Notebook/NotebookIconHeading.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const NotebookIconHeading = ({ level }: { level: number }): JSX.Element => {
```
Comment (benjackwhite) [inline frontend/src/scenes/notebooks/Notebook/NotebookIconHeading.tsx:1]: I think this is actually the way to go. Not sure we _need_ icons for stuff that is essentially text

File: frontend/src/scenes/notebooks/Notebook/NotebookIconHeading.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+    return (
+        <div style={{ fontSize: 16, fontWeight: '600' }}>
```
Comment (benjackwhite) [inline frontend/src/scenes/notebooks/Notebook/NotebookIconHeading.tsx:3]: eugh style...
```suggestion
        <div className="text-base font-semibold">
```

File: frontend/src/scenes/notebooks/Notebook/SlashCommands.tsx [hunk 3]
```diff
@@ -449,7 +457,7 @@ export const SlashCommands = forwardRef<SlashCommandsRef, SlashCommandsProps>(fu
     }
 
     return (
-        <div className="SlashCommands space-y-px">
+        <div className="space-y-1">
```
Comment (benjackwhite) [inline frontend/src/scenes/notebooks/Notebook/SlashCommands.tsx:460]: Why this change? With Menus, we typically only have a px space, not a 1 (0.25rem)
```suggestion
        <div className="space-y-px">
```

**Review by benjackwhite — CHANGES_REQUESTED** [review]
Small changes but overall - yep!

File: frontend/src/scenes/notebooks/Notebook/SlashCommands.tsx (hunk 3 above)
Comment (daibhin) [reply frontend/src/scenes/notebooks/Notebook/SlashCommands.tsx:460]: The vertical margin looked incredibly tight to me at 1px because of the hover state on the buttons. The horizontal padding was also different
|Before|After|
|----|----|
| Screenshot 2023-10-13 at 10 39 12 | Screenshot 2023-10-13 at 10 37 27 |

Comment (benjackwhite) [reply frontend/src/scenes/notebooks/Notebook/SlashCommands.tsx:460]: I'm still tempted to push back, mainly because we are doing something here different to everywhere else. 

If we agree it's too tight then fine - change it everywhere, but weird we would have more padding here than any other similar UI

Comment (daibhin) [reply frontend/src/scenes/notebooks/Notebook/SlashCommands.tsx]: Fair point. The thing for me here is that the horizontal divider isn't something we use in other similar UI (e.g. LemonMenu). Where we do use one (e.g. in the `NotebookSelectButton`) the spacing of everything is on a 0.25rem basis. Will fix forwards in that case rather than changing this.

Ultimately we should probably decide at a design system level what impact dividers have on a menu but that's for another day

**Review by benjackwhite — APPROVED** [review]
